pypsa/linopt.py

```diff
@@ -1159,12 +1159,17 @@ def run_and_read_xpress(
         m.setlogfile(solver_logfile)
 
     if warmstart:
-        m.readbasis(warmstart)
+        try:
+            m.readbasis(n.basis_fn)
+            logger.info("Model basis loaded")
+        except:
```

Comment:
It is not recommended to make a general expect statement. What error do we expect in case the try section fails? A `FileNotFoundError`?

Reply:
But this is a general pattern in `linopt`.